src/USER-PHONON/dynamical_matrix.h

```diff
@@ -0,0 +1,72 @@
+//
+// Created by charlie sievers on 6/21/18.
+//
+
+#ifdef COMMAND_CLASS
+
+CommandStyle(dynamical_matrix,DynamicalMatrix)
+
+#else
+
+#ifndef LMP_DYNAMICAL_MATRIX_H
+#define LMP_DYNAMICAL_MATRIX_H
+
+#include "pointers.h"
+
+namespace LAMMPS_NS {
+
+    class DynamicalMatrix : protected Pointers {
+    public:
+        DynamicalMatrix(class LAMMPS *);
+        virtual ~DynamicalMatrix();
+        void command(int, char **);
+        void setup();
+
+    protected:
+        int eflag,vflag;            // flags for energy/virial computation
+        int external_force_clear;   // clear forces locally or externally
+
+
+        int triclinic;              // 0 if domain is orthog, 1 if triclinic
+        int pairflag;
+
+        int pair_compute_flag;            // 0 if pair->compute is skipped
+        int kspace_compute_flag;          // 0 if kspace->compute is skipped
+
+        int nvec;                   // local atomic dof = length of xvec
+
+        void energy_force(int);
+        void force_clear();
+        virtual void openfile(const char* filename);
+
+    private:
+        void options(int, char **);
+        void calculateMatrix();
+        void writeMatrix();
+        void convert_units(const char *style);
+        void displace_atom(int local_idx, int direction, int magnitude);
+
+        double conversion;
+        double conv_energy;
+        double conv_distance;
+        double conv_mass;
+        double del;
+        int igroup,groupbit;
+        int scaleflag;
+        int me;
+        bigint dynlen;
+        double **dynmat;
```

Comment:
`dynmat` does not have to be a class member, and it does not have to be a full matrix but a 3 x dynlen size matrix, since the matrix is constructed 3 rows at a time.